addons/bus/models/bus.py

```diff
@@ -43,6 +46,25 @@ def channel_with_db(dbname, channel):
     return channel
 
 
+def get_notify_payloads(channels):
+    """
+    Generates the json payloads for the imbus NOTIFY.
+    Splits recursively payloads that are too large.
+
+    :param list channels:
+    :return: list of payloads of json dumps
+    :rtype: list[str]
+    """
+    if not channels:
+        return []
+    payload = json_dump(channels)
+    if len(channels) == 1 or len(payload.encode('utf-8')) < NOTIFY_PAYLOAD_MAX_LENGTH:
+        return [payload]
+    else:
+        return (get_notify_payloads(channels[:len(channels) // 2]) +
```

Comment:
Maybe worth adding some logging? I would assume it happens on a very exceptional basis, so logging might be useful.

Reply:
Thanks for the input, I've added a warning when we've split the payload with the number of resulting splits. The number of splits is correlated with the size of the initial payload and the frequency of `NOTIFY` queries.

Reply:
@pivi-odoo why does it deserve a warning ?

Reply:
@amigrave To be honest, I wasn't really sure to log it as `info` or `warning`. I've looked at other instances in the code and concluded with: "Out of normal behaviour that was handled gracefully", therefor a warning.

If you think it's not warranted, I can switch it to `info`, I don't have any preference, and wasn't sure how to classify it in the first place 🤷🏼

Reply:
@pivi-odoo I don't remember what was the issue of the customer who encountered this issue (do you?), so the real question is: if the customer contacts you to say that there are warnings in their odoo.log, will you say:

> there's an issue in place X or Y that you should fix (then ok for warning)

or

> naah, don't consider this warning, it's not important (then it should be an info)

Reply:
Thanks for the feedback, I hadn't thought about looking at it from this angle :)
I've switched it to an `info` log.